lib/active_shipping/package_item.rb

```diff
@@ -61,7 +60,7 @@ def kilograms(options = {})
 
     def attribute_from_metric_or_imperial(obj, klass, metric_unit, imperial_unit)
       if obj.is_a?(klass)
-        return value
+        return obj
```

Comment:
@kmcphillips  This was the bug here that was causing the strange behavior in the test below.